site/src/pages/tutorials/tutorial-blog-rest-step3.mdx

```diff
@@ -0,0 +1,168 @@
+      if (value == null) {
+          throw new IllegalArgumentException(field + " is missing!");
+      }
+      return value.textValue();
+    }
+  ```
+3. Customize the default `convertRequest()` method as follows.
+    ```java filename=BlogRequestConverter.java
+    @Override
+    public Object convertRequest(ServiceRequestContext ctx,
+      AggregatedHttpRequest request, Class<?> expectedResultType,
+      @Nullable ParameterizedType expectedParameterizedResultType)
+        throws Exception {
+      if (expectedResultType == BlogPost.class) {
+        final JsonNode jsonNode = mapper.readTree(request.contentUtf8());
+        final int id = idGenerator.getAndIncrement();
+        final String title = stringValue(jsonNode, "title");
+        final String content = stringValue(jsonNode, "content");
+        return new BlogPost(id, title, content); // Create an instance of BlogPost object
+      }
+      return RequestConverterFunction.fallthrough();
+    }
+  ```
+
+### Assign a request converter
+
+In this step, assign the [request converter we customized](#write-a-request-converter) to our service method. Annotate the service method with <type://@RequestConverter> and specify the <type://RequestConverterFunction> class as `BlogPostRequestConverter.class`.
+
+```java filename=BlogService.java highlight=2
+@Post("/blogs")
+@RequestConverter(BlogPostRequestConverter.class)
+public HttpResponse createBlogPost(BlogPost blogPost) throws JsonProcessingException {
+  // Implement blog service
+}
+```
+
+## 3. Implement service code
+
+When the request for creation is received, our request converter creates an instance of a blog post object for us. We want to save the blog post object in the map (`blogPosts`) created in the `BlogService` class.
+
+Let's store the blog post information in the map by adding line 4, in the `createBlogPost()` method.
+
+```java filename=BlogService.java highlight=4
+@Post("/blogs")
+@RequestConverter(BlogPostRequestConverter.class)
+public HttpResponse createBlogPost(BlogPost blogPost) throws JsonProcessingException {
+  blogPosts.put(blogPost.id(), blogPost);
+}
+```
+
+## 4. Return response
+
+Now, it's time to return a response to our client. As the response, on top of the information received, add the ID of the post, created time, plus modified time which would be identical to the created time.
+
+Let's a return a response for blog post creation:
+
+1. Specify JSON object keys for each member of our blog post object, by annotating the following methods with `@JsonProperty`.
+  ```java filename=BlogPost.java highlight=3,6,9,12,15
+  final class BlogPost {
+    ...
+    @JsonProperty("id")
+    int id() { return id; }
+
+    @JsonProperty("title")
+    String title() { return title; }
+
+    @JsonProperty("content")
+    String content() { return content; }
+
+    @JsonProperty("createdAt")
+    long createdAt() { return createdAt; }
+
+    @JsonProperty("modifiedAt")
+    long modifiedAt() { return modifiedAt; }
```

Comment:
Actually, these are not a Java standard getter style.
How about using `getXXX()` style? If so, you can also remove `@JsonProperty("XXX")` for simplicity's sake.

Reply:
Ah, I believe I'll need @minwoox 's comment on this, since the code samples in the tutorial will have to be identical to the service sample provided on Git.

Reply:
Yeah, this is Armeria style. Let's use `getXXX()` style and remove `@JsonProperty("XXX")` as well.
I will make changes to the sample code. Sorry about it. 😅

Reply:
@minwoox , @ikhoon 
In such case, I get an error for not having a serializer. Should I add `@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)` for the `BlogPost` class or use `@JsonSetter` for each `getOOO()`? Or is there any other alternatives I should use? (Sorry, I am not familiar with this area)

Reply:
Oops let me see what happens

Reply:
Ah we have to make it public. 😅 
```
public final class BlogPost {
  ...
  public int getId() {...}

  public String getTitle() {...}
  ...
```

Reply:
Works like a charm! Thank you
Done 8d694d264d77c7bded67373d06cd348835ff38f1